### ISSUE TITLE
Sanwa Supply JY-P76USV wrong controller mapping

### ISSUE THREAD

**Issue by pkr-sadx** [libsdl-org/SDL#8644]
Hi! I have a [Sanwa Supply JY-P76USV](https://www.sanwa.co.jp/product/syohin?code=JY-P76USV) controller that SDL identifies as GameShark GS-GP702. However, with SDL's Controller bindings the button locations are all wrong and some of the axes are unusable.

It's been like this for a while. In the last several years, the SDL controller database has had edits for this specific controller's entries, which sometimes made it work better, sometimes not. At the moment it's back to unusable without a custom controller map.

Here's the device GUID as reported by SDL 2.28.5: `0300f020790000000600000000000000`
In SDL 2.24.0 it reports a different GUID: `03000000790000000600000000000000`

Here's a mapping string that displays all inputs correctly in SDL2 Gamepad Mapper 0.0.5:
```
0300f020790000000600000000000000,Sanwa Supply JY-P76USV,a:b0,b:b1,x:b2,y:b3,back:b8,start:b9,leftshoulder:b4,rightshoulder:b5,leftstick:b10,rightstick:b11,dpup:h0.1,dpleft:h0.8,dpdown:h0.4,dpright:h0.2,-leftx:-a0,+leftx:+a0,-lefty:-a1,lefty:+a1,-rightx:-a2,+rightx:+a2,-righty:-a4,+righty:a4,lefttrigger:b6,righttrigger:b7,platform:Windows,
```

Here's what its button layout looks like when mapped to the X360 controller layout:
![image](https://github.com/libsdl-org/SDL/assets/147223739/2fff8870-33a7-483b-939d-18b0953c229e)
(triggers as buttons 7-8, shoulder buttons 5-6, axis and button indices are 1-based here)

It is possible to get this controller properly registered in the database? I'll be happy to provide any additional information.

EDIT: Device Manager info
Device instance path: `HID\VID_0079&PID_0006\6&3B6771A7&2&0000`
Hardware IDs:
```
HID\VID_0079&PID_0006&REV_0107
HID\VID_0079&PID_0006
HID\VID_0079&UP:0001_U:0004
```

**Comment by slouken** [comment]
Thank you for the mapping string, this has been added for the next release!

**Comment by PiKeyAr** [comment]
Hi! Thank you for the update.

Sorry to bump a closed issue. I just tested 2.30.1 with this controller and the mapping of the axes and buttons is now correct, but the centered values for Y axes are off. 

If I push Up or Down on the left stick all the way, the Y value goes to -32768 to 32767 respectively, but if I let it go, it stays at -32768. Pushing Up just a little bit from the centered position makes it jump from -32768 to -4800 or so, then it goes towards -32768 again as I push further. Pushing Down makes it go slowly from -32768 (centered) to 32767 (fully down) as I push further.

The right stick does the same thing except the centered value is 16383. Left and Right are fine on both sticks.

I'm not sure if this was happening on the previous version. I could've missed this when I made the mapping, so I apologize.

**Comment by slouken** [comment]
Yeah, it looks like that mapping string was incorrect.

Try this:
`"03000000790000000600000000000000,Sanwa Supply JY-P76USV,crc:20f0,a:b0,b:b1,back:b8,dpdown:h0.4,dpleft:h0.8,dpright:h0.2,dpup:h0.1,leftshoulder:b4,leftstick:b10,lefttrigger:b6,leftx:a0,lefty:+a1,lefty:a1,rightshoulder:b5,rightstick:b11,righttrigger:b7,rightx:a2,righty:a4,start:b9,x:b2,y:b3,"`

**Comment by PiKeyAr** [comment]
Thank you for sharing the mapping string. To test it, I removed the quotation marks and the last comma, appended `,platform:Windows,`  put it in `gamecontrollerdb.txt` and loaded it in my test program. Everything worked correctly.

**Comment by slouken** [comment]
Okay, this mapping has been updated for the next release, thanks!

**Comment by PiKeyAr** [comment]
Hi @slouken, sorry for the trouble but as of 2.30.6 this controller is no longer recognized as a GameController, at least on Windows. It seems that the GUID of the device reported by SDL has changed? The following mapping works correctly:
`03004f02790000000600000000000000,Sanwa Supply JY-P76USV,crc:20f0,a:b0,b:b1,back:b8,dpdown:h0.4,dpleft:h0.8,dpright:h0.2,dpup:h0.1,leftshoulder:b4,leftstick:b10,lefttrigger:b6,leftx:a0,lefty:a1,lefty:a1,rightshoulder:b5,rightstick:b11,righttrigger:b7,rightx:a2,righty:a4,start:b9,x:b2,y:b3,platform:Windows,`

**Comment by slouken** [comment]
This is fixed for the 2.32.0 release, can you please confirm in the 2.31.1 prerelease later today?

**Comment by PiKeyAr** [comment]
Hi! Sorry, I wasn't around...

I tested on 2.32.0 release and it seems the controller isn't being detected as a GameController just like on 2.30.6. Here's a screenshot from my test program:

![Image](https://github.com/user-attachments/assets/8172b019-bb4c-412c-bef1-1aaa9f1c5424)

The mapping that I provided [here](https://github.com/libsdl-org/SDL/issues/8644#issuecomment-2267084661) works fine if it's loaded manually from `gamecontrollerdb.txt`.

**Comment by slouken** [comment]
That's weird, because this is the mapping included in SDL:
```
03000000790000000600000000000000,Sanwa Supply JY-P76USV,crc:20f0,a:b0,b:b1,back:b8,dpdown:h0.4,dpleft:h0.8,dpright:h0.2,dpup:h0.1,leftshoulder:b4,leftstick:b10,lefttrigger:b6,leftx:a0,lefty:a1,lefty:a1,rightshoulder:b5,rightstick:b11,righttrigger:b7,rightx:a2,righty:a4,start:b9,x:b2,y:b3,
```

**Comment by slouken** [comment]
> It stops being detected if I change the GUID to that. I didn't know the GUID reported originally wasn't unique, sorry for the trouble. I think the VID and PID it reports aren't unique either, so I guess individual support for this controller would be impossible since it would clash with other generic controllers? It's no big deal, this controller is really old and I doubt anyone else has one... But thank you for looking into this!

That's correct. You're welcome! I'll go ahead and remove the incorrect mapping, since it seems like it isn't helpful.
